### PR TITLE
Add EVP demo for Poly1305 demonstrating Poly1305-AES

### LINKED ISSUE #14122: Add an EVP demo for POLY1305
It should be the current best practice (fetch), have proper error handling and resource allocation/deallocation and be in the style of a known answer test.

This is one of the PKEY legacy paths which are likely candidates for deprecation.

### PR DESCRIPTION
**Pull request by hlandau**
Fixes #14122.

### CODE REVIEW

**Review by paulidale — APPROVED** [review]

File: demos/mac/poly1305.c [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    if (!EVP_CIPHER_CTX_set_padding(aesctx, 0)) {
+        fprintf(stderr, "EVP_CIPHER_CTX_set_padding() failed\n");
+        goto end;
+    }
+
+    /*
+     * Computes the value AES_k(n) which we need for our Poly1305-AES
+     * computation below.
+     */
+    if (!EVP_EncryptUpdate(aesctx, composite_key+16, &aes_len, test_n, sizeof(test_n))) {
```
Comment (t8m) [inline demos/mac/poly1305.c:153]: Nit: spaces around `+`

**Review by t8m — CHANGES_REQUESTED** [review]
Would you please consider adding a Makefile similar to the one in cipher?

File: demos/mac/poly1305.c (hunk 1 above)
Comment (hlandau) [reply demos/mac/poly1305.c:153]: Fixed.

**Comment by hlandau** [comment]
@t8m Done.

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
@paulidale still OK?

**Review by paulidale — APPROVED** [review]
still good

**Comment by paulidale** [comment]
Merged to both, thanks.

**Pull request closed without merging**